packages/cdktf/lib/terraform-resource.ts

```diff
@@ -231,4 +252,64 @@ export class TerraformResource
       this.cdktfStack
     );
   }
+
+  private parentStackResourceTargets() {
+    return TerraformStack.of(this).moveTargets;
+  }
+
+  /**
+   * Moves this resource to the location denoted by moveTarget.
+   * @param moveTarget The previously set string moveTarget denoting the location of the move
+   * @param index Optional The index corresponding to the key the resource is to appear in the foreach of a resource to move to
+   */
+  public moveTo(moveTarget: string, index?: string | number) {
+    const stackMoveTargets = this.parentStackResourceTargets();
+    const resourceToMoveTo =
+      stackMoveTargets.getResourceAddressByTarget(moveTarget);
+    if (this.terraformResourceType !== resourceToMoveTo.terraformResourceType) {
+      throw new Error(
+        `The move target "${moveTarget}" corresponding to the resource of type ${resourceToMoveTo.terraformResourceType} to move to differs from the resource of type ${this.terraformResourceType} being moved from`
+      );
+    }
+    const movedToId = index
+      ? typeof index === "string"
+        ? `${this.terraformResourceType}.${resourceToMoveTo.friendlyUniqueId}["${index}"]`
+        : `${this.terraformResourceType}.${resourceToMoveTo.friendlyUniqueId}[${index}]`
+      : `${this.terraformResourceType}.${resourceToMoveTo.friendlyUniqueId}`;
+    const movedFromId = `${this.terraformResourceType}.${this.friendlyUniqueId}`;
+    this._moved = { to: movedToId, from: movedFromId, renamed: false };
+    // TODO: add validation of correct Terraform Version
+  }
+
+  /**
+   * Adds a string moveTarget to this resource
+   * @param moveTarget The string move target that corresponds to the address of this resource
```

Comment:
I'd adjust the comment a little to describe what this is good for. For example we could explain that this string is a user defined key that should be unique and will later be referenced in a `.moveTo()` call. I'd also give a mention of that other method this method is supposed to be used together with.

Reply:
same for the other comment of `moveTo`. It could mention `addMoveTarget` as the method via which the moveTarget string was previously defined.

Reply:
👍 included